openmc/plotter.py

```diff
@@ -540,3 +602,279 @@ def _calculate_xs_elem_mat(this, types, temperature=294., cross_sections=None,
+    average cross section over all angles.
+
+    Parameters
+    ----------
+    this : openmc.Element, openmc.Nuclide, or openmc.Material
+        Object to source data from
+    types : Iterable of values of PLOT_TYPES
+        The type of cross sections to calculate
+    orders : Iterable of Integral, optional
+        The scattering order or delayed group index to use for the
+        corresponding entry in types. Defaults to the 0th order for scattering
+        and the total delayed neutron data.
+    temperature : float, optional
+        Temperature in Kelvin to plot. If not specified, a default
+        temperature of 294K will be plotted. Note that the nearest
+        temperature in the library for each nuclide will be used as opposed
+        to using any interpolation.
+    cross_sections : str, optional
+        Location of MGXS HDF5 Library file. Default is None.
+    ce_cross_sections : str, optional
+        Location of continuous-energy cross_sections.xml file. Default is None.
+        This is used only for expanding an openmc.Element object passed as this
+    enrichment : float, optional
+        Enrichment for U235 in weight percent. For example, input 4.95 for
+        4.95 weight percent enriched U. Default is None
+        (natural composition).
+
+    Returns
+    -------
+    energy_grid : numpy.ndarray
+        Energies at which cross sections are calculated, in units of eV
+    data : numpy.ndarray
+        Cross sections calculated at the energy grid described by energy_grid
+
+    """
+
+    # Check types
+    cv.check_type('temperature', temperature, Real)
+    if enrichment:
+        cv.check_type('enrichment', enrichment, Real)
+    cv.check_iterable_type('types', types, string_types)
+
+    cv.check_type("cross_sections", cross_sections, str)
+    library = openmc.MGXSLibrary.from_hdf5(cross_sections)
+
+    if isinstance(this, (openmc.Nuclide, openmc.Macroscopic)):
+        mgxs = _calculate_mgxs_nuc_macro(this, types, library, orders,
+                                         temperature)
+    elif isinstance(this, (openmc.Element, openmc.Material)):
+        mgxs = _calculate_mgxs_elem_mat(this, types, library, orders,
+                                        temperature, ce_cross_sections,
+                                        enrichment)
+    else:
+        raise TypeError("Invalid type")
+
+    # Convert the data to the format needed
+    data = np.zeros((len(types), 2 * library.energy_groups.num_groups))
+    energy_grid = np.zeros(2 * library.energy_groups.num_groups)
+    i = 0
+    for g in range(library.energy_groups.num_groups):
+        energy_grid[i: i + 2] = library.energy_groups.group_edges[g: g + 2]
+        i += 2
+    # Ensure the energy will show on a log-axis by replacing 0s with a
+    # sufficiently small number
+    if energy_grid[0] <= 0.:
+        energy_grid[0] = _MIN_E
+
+    for line in range(len(types)):
+        i = 0
+        for g in range(library.energy_groups.num_groups):
```

Comment:
Ditto, this may be simplified as:

```python
for line in range(len(types)):
    for g in range(library.energy_groups.num_groups):
        data[g*2:g*2+2] = mgxs[line, g]
```